# DS-SF-36 | 09 | Linear Regression, Part 2 | Assignment | Starter Code

In [1]:
import os

import pandas as pd
pd.set_option('display.max_rows', 10)
pd.set_option('display.notebook_repr_html', True)
pd.set_option('display.max_columns', 10)

import statsmodels.api as sm
import statsmodels.formula.api as smf

import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')

import seaborn as sns

/Users/jacobwhite/anaconda/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


In [2]:
df = pd.read_csv(os.path.join('..', 'datasets', 'dataset-09-advertising.csv'))

In [3]:
df

TV  Radio  Newspaper  Sales
0    230.1   37.8       69.2   22.1
1     44.5   39.3       45.1   10.4
2     17.2   45.9       69.3    9.3
3    151.5   41.3       58.5   18.5
4    180.8   10.8       58.4   12.9
..     ...    ...        ...    ...
193   38.2    3.7       13.8    7.6
194   94.2    4.9        8.1    9.7
195  177.0    9.3        6.4   12.8
196  283.6   42.0       66.2   25.5
197  232.1    8.6        8.7   13.4

[198 rows x 4 columns]

We have 198 observations in this outlier-free dataset.  Each observation records the budget used to run ads on TV, on radio, and in the newspapers (in thousands of dollars) as well as the sale figures (in thousands of units).

Today, we are going to study if we can predict sales based on the amount of advertising.

> ## Question 1.  How are the different variables correlated?

In [ ]:
# TODO

Answer: TODO

> ## Question 2.  Draw a scatter plot between `TV` (horizontal axis) and `Sales` (vertical axis).

In [ ]:
# TODO

> ## Question 3.  Using _seaborn_, draw the same scatter plot along with a fitted line between `TV` and `Sales`.

- (http://seaborn.pydata.org/generated/seaborn.lmplot.html)

In [ ]:
# TODO

> ## Question 4.  Estimate a simple linear regression between `TV` (explanatory variable) and `Sales` (response variable).

In [ ]:
# TODO

> ## Question 5.  Interpret the coefficients.  Are they significant?  What equation did you derive?

Answer: TODO

> ## Question 6.  How are the residual distributed?  Do they follow the linear modeling assumptions?

In [ ]:
# TODO

Answer: TODO

> ## Question 7.  Estimate two new simple linear regression models: (1) `Sales` against `Radio` and (2) `Sales` against `Newspaper`.

In [ ]:
# TODO

In [ ]:
# TODO

> ## Question 8.  What can you conclude of these two new models against the first one?

Answer: TODO

> ## Question 9.  Run a multiple linear regression using `TV`, `Radio`, and `Newspaper` as explanatory variables.

In [ ]:
# TODO

> ## Question 10.  What are your findings?

Answer: TODO